src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -808,6 +874,168 @@ TryDropShardOutsideTransaction(OperationId operationId,
 }
 
 
+/*
+ * TryDropSubscriptionOutsideTransaction drops subscription with the given name on the
+ * subscriber node if it exists. Note that this doesn't drop the replication slot on the
+ * publisher node. The reason is that sometimes this is not possible. To known
+ * cases where this is not possible are:
+ * 1. Due to the node with the replication slot being down.
+ * 2. Due to a deadlock when the replication is on the same node as the
+ *    subscription, which is the case for shard splits to the local node.
+ *
+ * So instead of directly dropping the subscription, including the attached
+ * replication slot, the subscription is first disconnected from the
+ * replication slot before dropping it. The replication slot itself should be
+ * dropped using DropReplicationSlot on the source connection.
+ */
+static bool
+TryDropSubscriptionOutsideTransaction(char *subscriptionName,
+									  char *nodeName,
+									  int nodePort)
+{
+	int connectionFlags = OUTSIDE_TRANSACTION;
+	MultiConnection *connection = GetNodeUserDatabaseConnection(connectionFlags,
+																nodeName, nodePort,
+																CitusExtensionOwnerName(),
+																NULL);
+
```

Comment:
All of these queries should be setting `lock_timeout` to 1s to avoid distributed deadlocks (see TryDropShardOutsideTransaction) for more explanation.